src/main/java/com/dataloom/datastore/services/LinkingService.java

```diff
@@ -80,7 +83,7 @@ public UUID link( UUID linkedEntitySetId, Set<Map<UUID, UUID>> linkingProperties
 
         // Warning: We assume that the restrictions on links are enforced/validated as specified in LinkingApi. In
         // particular, from now on we work on the assumption that only identical property types are linked on.
-        initializeComponents( linkIndexedByEntitySets.keySet(), linkIndexedByPropertyTypes, linkIndexedByEntitySets );
+        initializeComponents( dsm.getLatestSyncId( linkIndexedByEntitySets.keySet() ), linkIndexedByPropertyTypes, linkIndexedByEntitySets );
```

Comment:
Updating the function to take in a map would be great - the reason is that the Blocker (not just the step in blocker that calls elastic search) needs the syncId (for example to load the data about each entity before performing the search). So it feels more natural for me to specify the SyncId already at this step.